src/ORM/EagerLoader.php

```diff
@@ -633,8 +633,12 @@ public function loadExternal(Query $query, StatementInterface $statement): State
         $driver = $query->getConnection()->getDriver();
         [$collected, $statement] = $this->_collectKeys($external, $query, $statement);
 
+        // TODO: The EagerLoader will have to be updated to use the rows fetched by
+        // StatementInterface::fetchAll() instead of relying on StatementInterface::rowCount()
+        // to get the rows count.
+
         // No records found, skip trying to attach associations.
-        if (empty($collected) && $statement->count() === 0) {
+        if (empty($collected) && $statement->rowCount() === 0) {
```

Comment:
I am surprised this doesn't break tests.

Reply:
That's because Sqlite's driver uses BufferedStatement who's rowCount() doesn't a fetchAll() and returns the results count. For other db's PDOStatement::rowCount() does return the correct count for SELECT queries.

Dropping BufferedStatement would require first updating the eagerloader and association code to work with either fetchAll()'s results or `ResultSet` instead of statements.